### PR TITLE
Code review 1 - Rahul Shakya

### PR DESCRIPTION
**Pull request by shakyarl**
- Analysis of the program
The program looks ready for the backend services to be flushed in, simple dto design and easy to understand. 

- Was the program available in UC Github on time?
The program was available in UC Github on time.
- Is the program easy for you, as an outsider, to understand?
Yes, it was quite easy to understand as an outsider, the class diagram was easy to follow up and the functions requirement was able to clarify the different functions that the program allows to do.
- Does the program compile?
Yes, the program compiles.
- Links to three commits you made to your group's repository during Sprint 1.

[Commit 1](https://github.com/AustinGarrison/Ent-App-Development/commit/ae17ad66dac5fc5f5cd4ac4ab5489ec2b6e15287)
I added the meta tag to make the web layout have a responsive web design viewport.

[Commit 2](https://github.com/AustinGarrison/Ent-App-Development/commit/16ee3986e887a9d79c249d56f5a72843c1c6fb72)
I changed the directory for the css and added all the css styling into the style.css file to remove inline css. Also, I checked with a previous code review to see if the directory handling was working, and it wasn't working properly for me. I also added the css style to the index page, and the topnav fragment created using bootstrap.

[Commit 3](https://github.com/AustinGarrison/Ent-App-Development/commit/6515de207aaf86b55f9e1c642257e472ad3d4380)
I used the bootstrap element to design the navbar instead of the header tag as the design document shows it will be used for multiple pages.

### CODE REVIEW

File: src/main/resources/templates/fragments/topnav.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<div th:fragment="navbar" class="top-nav">
+    <nav class="navbar navbar-expand-lg navbar-light">
+        <div class="container-fluid">
+            <h4>DnD Character Creator</h4>
+        </div>
+    </nav>
+</div>
```
Comment (shakyarl) [inline src/main/resources/templates/fragments/topnav.html:7]: Added in css elements not only for the index page but also the topnav fragment which will be reused for their 2 pages.

File: src/main/resources/templates/index.html [hunk 2]
```diff
@@ -1,149 +1,100 @@
 <!DOCTYPE html>
-<html lang="en" xmlns="http://www.w3.org/1999/html">
+<html lang="en" xmlns="http://www.w3.org/1999/html" xmlns:th="http://www.thymeleaf.org">
```
Comment (shakyarl) [inline src/main/resources/templates/index.html:2]: added in thymeleaf to make reusable fragment topnav

File: src/main/resources/templates/index.html [hunk 3]
```diff
@@ -1,149 +1,100 @@
...
 <head>
+    <!-- Required meta tags -->
     <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
```
Comment (shakyarl) [inline src/main/resources/templates/index.html:6]: added the metatag to improve web design responsiveness

File: src/main/resources/templates/index.html [hunk 4]
```diff
@@ -1,149 +1,100 @@
...
+
     <title>Character Creator</title>
-    <link rel="stylesheet" type="text/css" href="css/style.css">
+    <link rel="stylesheet" type="text/css" href="css/style.css" th:href="@{css/style.css}">
     <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC" crossorigin="anonymous">
```
Comment (shakyarl) [inline src/main/resources/templates/index.html:10]: added ref for css styling to be usable for the fragment created using Thymeleaf

File: src/main/resources/templates/index.html [hunk 5]
```diff
@@ -1,149 +1,100 @@
...
     <script src="https://code.jquery.com/jquery-3.6.0.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script>
     <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/js/bootstrap.bundle.min.js" integrity="sha384-MrcW6ZMFYlzcLA8Nl+NtUVF0sA7MsXsP1UyJoMp4YLEuNSfAP+JcXn/tWtIaxVXM" crossorigin="anonymous"></script>
 </head>
 
-<style>
```
Comment (shakyarl) [inline src/main/resources/templates/index.html:12]: remove inline css to a .css file in a directory for ease of usability with other pages and fragments

File: src/main/resources/static/css/style.css [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    color: white;
+    font-weight: bold;
+    font-size: 20px;
+    justify-content: center;
+    align-items: center;
+}
+
+.container-fluid h4{
+    text-align: center;
+}
```
Comment (shakyarl) [inline src/main/resources/static/css/style.css:58]: Added in css elements to the css file not only for the index page but also the topnav fragment which will be reused for their 2 pages. Adding styling to a css file makes it more accessible to more pages, and results to less typing. Also, file directory is different compared to a previous code reviewer, as I checked if the styling would work, and it didn't work properly for me through the previous file directory handling.

File: src/main/resources/templates/index.html (hunk 5 above)
Comment (discospiff) [reply src/main/resources/templates/index.html:12]: Good idea.l

**Comment by discospiff** [comment]
Moving inline styles to an external stylesheet is a good idea.
Other changes: the group should review and decide whether or not to implement.

**Pull request closed without merging**